templates.go

```diff
@@ -529,13 +529,13 @@ coreos:
       ExecStartPre=/bin/bash -c "while [ ! -f /etc/kubernetes/ssl/etcd/server-ca.pem ]; do echo 'Waiting for /etc/kubernetes/ssl/etcd/server-ca.pem to be written' && sleep 1; done"
       ExecStartPre=/bin/bash -c "while [ ! -f /etc/kubernetes/ssl/etcd/server-crt.pem ]; do echo 'Waiting for /etc/kubernetes/ssl/etcd/server-crt.pem to be written' && sleep 1; done"
       ExecStartPre=/bin/bash -c "while [ ! -f /etc/kubernetes/ssl/etcd/server-key.pem ]; do echo 'Waiting for /etc/kubernetes/ssl/etcd/server-key.pem to be written' && sleep 1; done"
-      ExecStart=/usr/bin/etcd2 --advertise-client-urls=https://0.0.0.0:2379 \
+      ExecStart=/usr/bin/etcd2 --advertise-client-urls=https://{{ .Cluster.Etcd.Domain }}:2379 \
                                --data-dir=/etc/kubernetes/data/etcd/ \
-                               --initial-advertise-peer-urls=https://0.0.0.0:2380 \
+                               --initial-advertise-peer-urls=https://{{ .Cluster.Etcd.Domain }}:2380 \
```

Comment:
OK, it works :+1: 

```
# /usr/bin/etcd2 --advertise-client-urls=https://etcd.michal.aws.giantswarm.io:2379                          --data-dir=/etc/kubernetes/data/etcd/                          --initial-advertise-peer-urls=https://127.0.0.1:2380                          --listen-client-urls=https://0.0.0.0:2379                          --listen-peer-urls=https://${DEFAULT_IPV4}:2380                          --initial-cluster-token k8s-etcd-cluster                          --initial-cluster etcd0=https://127.0.0.1:2380                          --initial-cluster-state new                          --ca-file=/etc/kubernetes/ssl/etcd/server-ca.pem                          --cert-file=/etc/kubernetes/ssl/etcd/server-crt.pem                          --key-file=/etc/kubernetes/ssl/etcd/server-key.pem                          --peer-ca-file=/etc/kubernetes/ssl/etcd/server-ca.pem                          --peer-cert-file=/etc/kubernetes/ssl/etcd/server-crt.pem                          --peer-key-file=/etc/kubernetes/ssl/etcd/server-key.pem                          --peer-client-cert-auth=true                          --name etcd0
2017-04-26 15:45:36.241878 I | etcdmain: etcd Version: 2.3.7
2017-04-26 15:45:36.241938 I | etcdmain: Git SHA: fd17c91
2017-04-26 15:45:36.241942 I | etcdmain: Go Version: go1.7.5
2017-04-26 15:45:36.241946 I | etcdmain: Go OS/Arch: linux/amd64
2017-04-26 15:45:36.241950 I | etcdmain: setting maximum number of CPUs to 2, total number of available CPUs is 2
2017-04-26 15:45:36.242007 N | etcdmain: the server is already initialized as member before, starting as etcd member...
2017-04-26 15:45:36.242031 I | etcdmain: peerTLS: cert = /etc/kubernetes/ssl/etcd/server-crt.pem, key = /etc/kubernetes/ssl/etcd/server-key.pem, ca = /etc/kubernetes/ssl/etcd/server-ca.pem, trusted-ca = , client-cert-auth = true
2017-04-26 15:45:36.242701 I | etcdmain: listening for peers on https://:2380
2017-04-26 15:45:36.242722 I | etcdmain: clientTLS: cert = /etc/kubernetes/ssl/etcd/server-crt.pem, key = /etc/kubernetes/ssl/etcd/server-key.pem, ca = /etc/kubernetes/ssl/etcd/server-ca.pem, trusted-ca = , client-cert-auth = false
2017-04-26 15:45:36.243367 I | etcdmain: listening for client requests on https://0.0.0.0:2379
2017-04-26 15:45:36.244205 I | etcdserver: name = etcd0
2017-04-26 15:45:36.244224 I | etcdserver: data dir = /etc/kubernetes/data/etcd/
2017-04-26 15:45:36.244231 I | etcdserver: member dir = /etc/kubernetes/data/etcd/member
2017-04-26 15:45:36.244241 I | etcdserver: heartbeat = 100ms
2017-04-26 15:45:36.244248 I | etcdserver: election = 1000ms
2017-04-26 15:45:36.244255 I | etcdserver: snapshot count = 10000
2017-04-26 15:45:36.244273 I | etcdserver: advertise client URLs = https://etcd.michal.aws.giantswarm.io:2379
2017-04-26 15:45:36.252244 I | etcdserver: restarting member 48aea1ad8789a678 in cluster 147e329f2adb9a7d at commit index 3266
2017-04-26 15:45:36.252450 I | raft: 48aea1ad8789a678 became follower at term 4
2017-04-26 15:45:36.252469 I | raft: newRaft 48aea1ad8789a678 [peers: [], term: 4, commit: 3266, applied: 0, lastindex: 3266, lastterm: 4]
2017-04-26 15:45:36.254900 I | etcdserver: starting server... [version: 2.3.7, cluster version: to_be_decided]
2017-04-26 15:45:36.255088 E | etcdmain: failed to notify systemd for readiness: No socket
2017-04-26 15:45:36.255102 E | etcdmain: forgot to set Type=notify in systemd service file?
2017-04-26 15:45:36.255611 N | etcdserver: added local member 48aea1ad8789a678 [https://0.0.0.0:2380] to cluster 147e329f2adb9a7d
2017-04-26 15:45:36.255680 N | etcdserver: set the initial cluster version to 2.3
2017-04-26 15:45:37.553903 I | raft: 48aea1ad8789a678 is starting a new election at term 4
2017-04-26 15:45:37.554094 I | raft: 48aea1ad8789a678 became candidate at term 5
2017-04-26 15:45:37.554194 I | raft: 48aea1ad8789a678 received vote from 48aea1ad8789a678 at term 5
2017-04-26 15:45:37.554285 I | raft: 48aea1ad8789a678 became leader at term 5
2017-04-26 15:45:37.554355 I | raft: raft.node: 48aea1ad8789a678 elected leader 48aea1ad8789a678 at term 5
2017-04-26 15:45:37.556721 I | etcdserver: published {Name:etcd0 ClientURLs:[https://etcd.michal.aws.giantswarm.io:2379]} to cluster 147e329f2adb9a7d
```

So, if you agree with that approach, I will make a change.